xml_converter/generators/code_generator.py

```diff
@@ -359,13 +388,55 @@ def build_example(self, type: str, applies_to: List[str], xml_field: str, exampl
 
         return example
 
+    ############################################################################
+    # Generate Node Types
+    #
+    # This will output code for a single category of nodes.
+    ############################################################################
+    def generate_node_types(self, metadata: Any, attributenames: Dict[str,str],page: str) -> Tuple[List[str],List[str]]:
+        
+        field_rows: List[str] = []
+        includelist: Set[str] = set()
+        typechange: Dict[str,str] = {
+            "Fixed32": "int",
+            "Int32": "int",
+            "Boolean": "bool",
+            "Float32": "float",
+            "String": "string",
+        }
+        
+        for fieldkey,field in metadata.items():
+            for x in attributenames:
+                if fieldkey in x :
+                    attributename = attributenames[x]
```

Comment:
It is possible `attributename` will not get set here if `attributenames` is an empty list, or if `fieldkey in x` is never true. This means that later when `attributename` is trying to be used it wont have yet been defined.

Reply:
updated to `attribute_names: Dict[str,str] = {}`